### ISSUE TITLE
False positive in Probable bugs > Suspicious assignments (use 'else' here)

### ISSUE THREAD

**Issue by fraserredmond** [kalessil/phpinspectionsea#1406]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 3.0.15 |
| Language level | PHP 7.3                        |
| PhpStorm  | 2019.2.1                        |

Gets a warning `immediately overridden, perhaps it was intended to use 'else' here.` with this code:
```
<?php
declare(strict_types=1);
$out = '';
if (true) { // Remove these 2 lines (and/or everthing inside) and the error goes away. 
    $variableGetsError    = '';
    if (true) {
        $variableGetsError .= 'bar';
    }
    $out .= $variableGetsError;
} // Remove these 2 lines (and/or everthing inside) and the error goes away. 
$variableGetsError    = ''; // ERROR ON THIS LINE
$variableGetsError .= 'theEnd';
$out .= $variableGetsError;
echo $out;
```
If you take out the 2 lines, or the whole of the if-block then the warning goes away.

**Comment by kalessil** [comment]
Uh, that's a challenge... I'll check if we can reliably identify the variables declaration scope...

**Comment by fraserredmond** [comment]
Thanks. I forgot to say, but it happens inside a function too.

**Comment by kalessil** [comment]
Giving up, I can not find reliable enough solution for this case, sorry.

**Comment by fraserredmond** [comment]
ok :(

Thanks for trying.
